Replaced io/ioutil with "os / io" package.

yanggangtony opened the pull request:
Signed-off-by: yanggang <gang.yang@daocloud.io>

/kind cleanup
Hey , calico community..
We should clean all io/util package according the url: [ioutil deprecation](https://go.dev/doc/go1.16#ioutil)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/projectcalico/calico?pullRequest=7013) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/projectcalico/calico?pullRequest=7013) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/projectcalico/calico?pullRequest=7013) it.</sub>

caseydavenport reviewed (approved):
LGTM - thanks for doing this!

mgleung commented:
@yanggangtony it looks like you need to rebase before we can merge this.

marvin-tigera commented:
Removing "merge-when-ready" label due to new commits

yanggangtony commented:
@mgleung 
Hi, I had rebase it ..
sorry for that forget tell you :grinning::grinning:

caseydavenport commented:
/sem-approve

mgleung commented:
@yanggangtony sorry, it looks like this needs another rebase since I didn't see this in time. Could I ask you to rebase one more time?

marvin-tigera commented:
Removing "merge-when-ready" label due to new commits

yanggangtony commented:
> @yanggangtony sorry, it looks like this needs another rebase since I didn't see this in time. Could I ask you to rebase one more time?

:grimacing::grimacing:
Rebased again.

caseydavenport commented:
/sem-approve

yanggangtony commented:
<img width="1277" alt="image" src="https://user-images.githubusercontent.com/12080746/211422056-f92ab611-a0a0-4cec-b5d9-7ec75126c435.png">
<img width="1231" alt="image" src="https://user-images.githubusercontent.com/12080746/211422170-89f6596c-a9f0-42a6-963d-52d358cd2668.png">

@caseydavenport 
Like above output , i execute the 'make generate' , its ok , but has no funcstion to the current code .
its meanings there is no change?
you means 'out of date' in the code? but when i rebase the master , it will follow the latest code

mgleung commented:
@yanggangtony I suspect that all of the changes that our checks are picking up have to do with updates to the year. After you run `make generate` on your code, what does `git diff` show? According to our automation, you should have a bunch of 2 line changes (which I suspect are just updates to the new year).

yanggangtony commented:
<img width="359" alt="image" src="https://user-images.githubusercontent.com/12080746/211480126-f5387b91-86f8-4108-9435-792c0319a55d.png">

@mgleung 
Hi, thanks for review.
I execute 'git status' and 'git diff' , there is no change codes.:smile::smile:

caseydavenport commented:
@yanggangtony could you try rebasing on master and running `make generate` one more time? Curious if we've fixed the issue in master or not.

marvin-tigera commented:
Removing "merge-when-ready" label due to new commits

yanggangtony commented:
> @yanggangtony could you try rebasing on master and running `make generate` one more time? Curious if we've fixed the issue in master or not.

```
	modified:   .semaphore/semaphore-scheduled-builds.yml
	modified:   .semaphore/semaphore.yml
```
Rebased , And execute 'make generate'.. shows the before file is changed..

mgleung commented:
/sem-approve

yanggangtony commented on .semaphore/semaphore-scheduled-builds.yml:
```diff
@@ -102,7 +102,7 @@ blocks:
 
 - name: "API"
   run:
-    when: "true or change_in(['/*', '/api/'], {exclude: ['/**/.gitignore', '/**/README.md', '/**/LICENSE']})"
```
@caseydavenport 
you mean just the line 105 is wrong?
or others is the same problem?
Should i look the history of the '.semaphore/semaphore-scheduled-builds.yml' , and rebase it?

yanggangtony replied:
> 

reset it.

caseydavenport commented:
/sem-approve

mgleung commented:
@yanggangtony I hate to have to say this again, but could you run `make generate` again? It looks like some of the operator CRDs have changed that we need to pick up.

hjiawei commented:
@mgleung Rebasing will also do the trick as this branch is a just a few commits behind (`operator-crds.yaml` dirty issue seems to be fixed on master). However, as this changeset touches felix/node and their [UTs are failing](https://tigera.semaphoreci.com/workflows/e001c820-183b-42fb-a42c-780166d1004a/summary?pipeline_id=080fb274-00b1-4936-8611-2fa2f5890d79&report_id=18bd7c02-4d8d-3cf1-acc4-829a795ceeb5) on master. It is unlikely for this to pass the build. As this PR has been approved (several times), I would suggest get this force merged.

mgleung commented:
@hjiawei the failures in the BPF tests and the Felix FVs I think we can navigate. I think I am less comfortable with force merging this since in all of the tests for the history of the PR, we have either had issues with the static checks (for building Felix), and after that, failures with the pre-flight checks which have prevented us from seeing the results of the fixes for the Felix build. If we were hitting some of the test failures we were aware of, I would be fine, but I want to get a build running where everything has at least been built before we merge the changes.

mgleung commented:
@yanggangtony as per @hjiawei 's suggestion, could I ask you to rebase this really quick? I imagine we won't see any conflicts this time and that will hopefully fix the tests up enough where everything will build.

yanggangtony commented:
rebase it.

caseydavenport commented:
/sem-approve

caseydavenport commented on felix/bpf/ut/bpf_prog_test.go:
```diff
@@ -227,7 +226,20 @@ func setupAndRun(logger testLogger, loglevel, section string, rules *polprog.Rul
+	maps := make([]bpf.Map, len(progMaps))
+	copy(maps, progMaps)
+
+outer:
```
Is this change related to the os/io replacement?

yanggangtony replied:
Ah, these code was few days ago , the code was conflict.
But i see the master branch , have no that code .
i will delete that..

caseydavenport commented:
/sem-approve

caseydavenport commented:
Thanks for sticking with this one @yanggangtony !

yanggangtony commented:
> Thanks for sticking with this one @yanggangtony !

So long time for yours reviews..
thanks very much..
